src/app/components/wrapper-components/part-creation/part-creation.component.ts

```diff
@@ -310,99 +337,123 @@ export class PartCreationComponent implements OnInit, OnDestroy {
     private observeNeededPlayers(): void {
         const joiner: Joiner = Utils.getNonNullable(this.currentJoiner);
         display(PartCreationComponent.VERBOSE, { PartCreationComponent_updateJoiner: { joiner } });
-        if (this.userName === joiner.creator) {
+        if (this.authUser.userId === joiner.creator.id) {
             this.observeCandidates();
         } else {
             this.observeCreator();
         }
     }
     private observeCreator(): void {
         const joiner: Joiner = Utils.getNonNullable(this.currentJoiner);
-        if (this.creatorSubscription != null) {
-            // We are already observing the creator
+        this.observeUserIfNotDone(joiner.creator.id);
+    }
+    private async destroyDocIfPartDidNotStart(): Promise<void> {
+        const joiner: Joiner = Utils.getNonNullable(this.currentJoiner);
+        const partStatusStarted: boolean = joiner.partStatus === PartStatus.PART_STARTED.value ||
+                                           joiner.partStatus === PartStatus.PART_FINISHED.value;
+        assert(partStatusStarted === false, 'Should not try to cancelGameCreation when part started!');
+        assert(this.allDocDeleted === false, 'Should not delete doc twice');
+        await this.cancelGameCreation();
+    }
+    /**
+     * Set the user doc subscription, which will set timeout for user (then resetting it while present)
```

Comment:
which will set *the* timeout for *the* (current?) user

Reply:
Tout se fait dès qu'on reçois un changement de l'user (ou le premier doc si c'est le premier abonnement à celui-ci) dans onUserUpdate, c'est là que le [re]set est fait. C'est dans une fonction appellée par la fonction commentée que l'évènement du commentaire arrive, mais c'est une sous-fonction B privée utilisée uniquement  dans cette fonction A.
Ca a un nom une fonction comme ça d'ailleurs ?

Reply:
une méthode privée simplement ? (si onUserUpdate serait définie dans A, alors ça serait une fonction locale, mais c'po lco) Mais ok du coup ! (tu peux résoudre)

Reply:
une fonction privée peut être utilisée par deux fonctions, quand elle l'est que par un je voulais dire :p